### ISSUE TITLE
Check-In Manager fails to load when shared child GroupType is used

### ISSUE THREAD

**Issue by jbeveland27** [SparkDevNetwork/Rock#4636]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * ✅  Did you perform a search at https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock to see if your bug or enhancement is already reported?
    * ❌ **Did not attempt** ~Can you reproduce the problem on a fresh install or the [demo site](http://rock.rocksolidchurchdemo.com/)?~
    * ✅  Did you include your Rock version number and [client culture](https://github.com/SparkDevNetwork/Rock/wiki/Environment-and-Diagnostics-Information) setting?

### Description

Experiencing an error loading the Check-In Manager. Unfortunately I don't have a capture of the screen that was happening, as we've fixed this temporarily by removing the offending GroupType from being an `Allowed Child Group Type` in multiple check-in configs.

![image](https://user-images.githubusercontent.com/6722433/112209637-0eb3ac00-8bf0-11eb-835a-e95d08d3a246.png)

Stack Trace: 

```
System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable1 source, Func2 keySelector, Func2 elementSelector, IEqualityComparer1 comparer)
at Rock.CheckIn.RosterAttendee.GetFromAttendanceList(IList`1 attendanceList) in C:\projects\Rock\Rock\CheckIn\RosterAttendee.cs:line 588
at RockWeb.Blocks.CheckIn.Manager.Roster.BindGrid()
at RockWeb.Blocks.CheckIn.Manager.Roster.BuildRoster()
at RockWeb.Blocks.CheckIn.Manager.Roster.OnLoad(EventArgs e)
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Control.LoadRecursive()
at System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)
```

We have a staff group type that has permission to be a check-in area (i.e. `Allowed Child Group Type`) under multiple check-in configs. When [GetAllCheckinAreaPaths](https://github.com/SparkDevNetwork/Rock/blob/hotfix-1.12.2/Rock/CheckIn/RosterAttendee.cs#L588) is called, it grabs all descendant GroupTypes under that root config GroupTypeId here: https://github.com/SparkDevNetwork/Rock/blob/hotfix-1.12.2/Rock/Model/GroupTypeService.Partial.cs#L128

Since our staff groupType would be returned under multiple check-in configs, this causes an issue when constructing the Dictionary: https://github.com/SparkDevNetwork/Rock/blob/hotfix-1.12.2/Rock/CheckIn/RosterAttendee.cs#L588

### Steps to Reproduce

1. Create a new GroupType called `Test`
2. Go to multiple Check-In Template GroupTypes, and add the `Test` GroupType under `General` > `Allowed Child Group Types`.
3. Go to Check-In Manager and try to load the Room Manager.

### Versions

* **Rock Version:** Rock McKinley 12.2 (1.12.2.2)
* **Client Culture Setting:** en-US

cc: @adamhann

**Comment by leahjennings** [comment]
We are also running 12.2 and get the same error when accessing the Room List page in Check-in Manager. I haven't been able to workaround it because I haven't figured out which group type is the offender. If any tests or screenshots need to be grabbed, I'd be happy to provide them from our system.

**Comment by nairdo** [comment]
@leahjennings / @jbeveland27  I'm trying to track it down, but this might have been one of the items we corrected in v12.3.  Are either of you in a position to be able to alpha test version 12.3 (which will be happening very soon)?

**Comment by leahjennings** [comment]
@nairdo let me see what I can do! We have a dev environment, I just don't access it that much, but I will try and get it updated to alpha test 12.3 and report back!

**Comment by adamhann** [comment]
@nairdo I was just chatting with @jbeveland27 and we'll try to update our Dev environment to 12.3 later today and test.

**Comment by jbeveland27** [comment]
@nairdo We manage a fork and build from source. I was planning to pull `hotfix-1.12.3` into a branch today in our repo and deploy that to our Dev. Does that sound like it'd be a good smoke test for what's in-progress or should we target a different branch? Or just wait till Friday/Monday?

**Comment by nairdo** [comment]
> Does that sound like it'd be a good smoke test...

Yes, that would work well.

**Comment by adamhann** [comment]
@nairdo Just reporting back. We tested Check-In Manager in 12.3 Alpha. We can load the Room Manager and see the Room Roster now while having a Group Type in multiple Check-in Templates. 

With that said, I checked in a few people and they do not show up on the Room Roster. I can see them under the Live Metrics, but not under Room Roster. This may be because this is still in Alpha. Let us know if there's anything else we can test.

**Comment by nairdo** [comment]
> ...a few people and they do not show up on the Room Roster. ...This may be because this is still in Alpha. 

@adamhann  They really should be showing up in the Room Roster if you have the correct room/location selected that they are checked into.  Also, depending on whether you have "Presence" and "Check-out" enabled, they could show up in different tabs of the Roster -- depending on their 'state'.

**Comment by cwomack** [comment]
@adamhann It looks like the main concern of the Check-in Manager failing to load will be fixed in 12.3, but there's a different issue with the Room Roster not showing people you've checked in.  Would you mind opening a separate issue tied to that with any data/configuration/steps to reproduce you have so we can target what's happening? 

I'll close this issue out so we don't mix the two.
